php-checks/src/main/java/org/sonar/php/checks/ClassCouplingCheck.java

```diff
@@ -158,38 +161,25 @@ private void retrieveTypeFromCommentLine(String line) {
     }
   }
 
-  private void retrieveInstantiatedClassName(AstNode astNode) {
-    String className = getInstantiatedClassName(astNode);
-    if (className != null) {
-      types.add(className);
-    }
-  }
+  private void retrieveInstantiatedClassName(NewExpressionTree newExpression) {
+    if (newExpression.expression().is(Kind.FUNCTION_CALL)) {
+      ExpressionTree callee = ((FunctionCallTree) newExpression.expression()).callee();
 
-  private String getInstantiatedClassName(AstNode newExpr) {
-    AstNode variable = newExpr.getFirstChild(PHPGrammar.MEMBER_EXPRESSION);
-    AstNode classNameNode = variable.getFirstChild();
-
-    if (classNameNode.is(PHPKeyword.NAMESPACE)) {
-      return getClassName(variable);
-    } else if (classNameNode.is(PHPGrammar.CLASS_NAME, PHPGrammar.IDENTIFIER) && classNameNode.getFirstChild().isNot(PHPKeyword.STATIC)) {
-      return getClassName(classNameNode);
-    } else {
-      return null;
+      if (callee.is(Kind.NAMESPACE_NAME)) {
+        types.add(getTypeName(((NamespaceNameTree) callee)));
+      }
     }
   }
 
-  private String getClassName(AstNode expr) {
-    StringBuilder builder = new StringBuilder();
-
-    for (Token token : expr.getTokens()) {
-      if (token.getType() == PHPPunctuator.LPARENTHESIS) {
-        break;
-      }
-      if (token.getType() != PHPKeyword.NAMESPACE) {
-        builder.append(token.getOriginalValue());
-      }
+  private String getTypeName(NamespaceNameTree namespaceName) {
+    String name = namespaceName.fullName();
+    String prefix = "namespace\\";
+    if (StringUtils.startsWithIgnoreCase(name, prefix)) {
+      // fixme (SONARPHP-552): Handle namespaces properly
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)